### PR TITLE
Bump cross-arm-none-eabi gcc and binutils to match non-cross packages

### PR DESCRIPTION
**Pull request by via**
#### Testing the changes
- I tested the changes in this PR: YES
I tested this with my own personal armv7 projects.  Notably, the version currently in packages has a bug with LTO that makes it discard ISR functions/symbols inappropriately, making LTO useless for some embedded use cases.  This appears fixed in gcc 10.

#### Local build testing
- I built this PR locally for my native architecture, amd64/glibc

### CODE REVIEW

**Comment by JamiKettunen** [comment]
Hm this is not being updated in #34902 unlike `cross-arm-linux-gnueabi` and `cross-arm-linux-musleabi`, I wonder if there is a particular reason why

**Comment by JamiKettunen** [comment]
Copying @oreo639's response from `#xbps`:
> The cross-*-linux packages are dependant on the regular gcc/binutils/libc packages and pull patches from them, meaning that they have to be kept in sync (not to mention not updating them would cause inconsistent behavior in the builders when cross compiling vs native compiling).
>
> This isn't the case with cross-arm-none-eabi which is its own thing (generic newlib gcc compiler), meaning it makes more sense for it to be updated seperately. (preferably by someone who actually does embedded stuff and uses it)

**Comment by via** [comment]
Hi, I definitely still have interest in this, and use this set of packages for embedded use, I just haven't had time to fix the issues that cropped up.  Additionally I think I'd need to bump libstdc++ too, and at this point I might as well match the versions that gcc/binutils uses now.  I'll take another stab at this soon.

**Comment by JamiKettunen** [comment]
> We are actively working on updating gcc here: #34902

@paper42 read my message a bit above, it excludes `arm-none-eabi` stuff entirely as this compiler is not used to build packages.

File: srcpkgs/cross-arm-none-eabi-binutils/template [hunk 1]
```diff
@@ -30,7 +30,7 @@ maintainer="Ivan Sokolov <ivan-p-sokolov@ya.ru>"
 license="GPL-3.0-or-later"
 homepage="https://www.gnu.org/software/${_pkgname}/"
 distfiles="${GNU_SITE}/${_pkgname}/${_pkgname}-${version}.tar.xz"
-checksum=0ab6c55dd86a92ed561972ba15b9b70a8b9f75557f896446c82e8b36e473ee04
+checksum=1b11659fb49e20e18db460d44485f09442c8c56d5df165de9461eb09c8302f85
 nocross=yes
```
Comment (wryun) [inline srcpkgs/cross-arm-none-eabi-binutils/template:34]: Random drive-by query - why is this nocross=yes (cross-arm-none-eabi-gdb/newlib don't have this)?

**Comment by riddicc** [comment]
> bump it

bump! 😋
